### PR TITLE
Add "plugin" support to local backend

### PR DESCRIPTION
**Pull request by 6543**
and a nit to improve the clone

---
*Sponsored by Kithara Software GmbH*

### CODE REVIEW

File: pipeline/backend/local/clone.go [hunk 1]
```diff
@@ -64,17 +63,20 @@ func (e *local) setupClone(state *workflowState) error {
 
 // execClone executes a clone-step locally
 func (e *local) execClone(ctx context.Context, step *types.Step, state *workflowState, env []string) error {
-	if err := e.setupClone(state); err != nil {
-		return fmt.Errorf("setup clone step failed: %w", err)
+	if scm := step.Environment["CI_REPO_SCM"]; scm != "git" {
+		return fmt.Errorf("local backend can only clone from git repos")
```
Comment (6543) [inline pipeline/backend/local/clone.go:67]:
```suggestion
		return fmt.Errorf("local backend can only clone from git repos, but this repo use '%s'", scm)
```

File: pipeline/backend/local/local.go [hunk 2]
```diff
@@ -148,6 +151,26 @@ func (e *local) execCommands(ctx context.Context, step *types.Step, state *workf
 	return cmd.Start()
 }
 
+// execPlugin use step.Image as exec binary
+func (e *local) execPlugin(ctx context.Context, step *types.Step, state *workflowState, env []string) error {
+	binary, err := exec.LookPath(step.Image)
+	if err != nil {
+		return fmt.Errorf("exec plugin: %w", err)
```
Comment (6543) [inline pipeline/backend/local/local.go:158]:
```suggestion
		return fmt.Errorf("lookup plugin binary: %w", err)
```

File: pipeline/backend/local/clone.go [hunk 3]
```diff
@@ -64,17 +63,20 @@ func (e *local) setupClone(state *workflowState) error {
...
 	}
 
-	if step.Image != constant.DefaultCloneImage {
-		// TODO: write message into log
-		log.Warn().Msgf("clone step image '%s' does not match default git clone image. We ignore it assume git.", step.Image)
+	if err := e.setupClone(state); err != nil {
+		return fmt.Errorf("setup clone step failed: %w", err)
+	}
+
+	if !strings.Contains(step.Image, "woodpeckerci/plugin-git") {
```
Comment (6543) [inline pipeline/backend/local/clone.go:78]:
```suggestion
	if !strings.Contains(step.Image, "plugin-git") {
```

File: docs/docs/30-administration/22-backends/20-local.md [hunk 4]
```diff
@@ -92,6 +88,17 @@ steps:
...
 
+### Plugins as Executable Binaries
+
+```yaml
+steps:
+  build:
+    image: /usr/bin/tree
+```
+
+If no commands are provided, we treat them as plugins in the usual manner.
```
[the remark this answers is not in the document]
Comment (6543) [reply docs/docs/30-administration/22-backends/20-local.md:99]: yes two spaces at the end means a newline

tldr: `  ` = `\n`

File: docs/docs/30-administration/22-backends/20-local.md [hunk 5]
```diff
@@ -92,6 +88,17 @@ steps:
       [...]
 ```
 
+### Plugins as Executable Binaries
+
+```yaml
+steps:
+  build:
+    image: /usr/bin/tree
```
Comment (anbraten) [inline docs/docs/30-administration/22-backends/20-local.md:96]: As this is not an image should we consider adding a different property maybe to stay clean?

Comment (6543) [reply docs/docs/30-administration/22-backends/20-local.md:96]: I would say stay clean as it is now ...

the local backend is not the main focus of woodpecker ... it's just to cover all edgecases
